libs/coin-modules/coin-solana/src/hw-signMessage.ts

```diff
@@ -0,0 +1,30 @@
+import { SignerContext } from "@ledgerhq/coin-framework/signer";
+import { Account, AnyMessage, DefaultMessage, DeviceId } from "@ledgerhq/types-live";
+import { SolanaSigner } from "./signer";
+
+export const signMessage =
+  (signerContext: SignerContext<SolanaSigner>) =>
+  async (
+    deviceId: DeviceId,
+    account: Account,
+    messageOptions: AnyMessage,
+  ): Promise<{
+    signature: string;
+  }> => {
+    if (!messageOptions.message || typeof messageOptions.message !== "string") {
+      throw new Error(
+        "Sign off-chain message on Solana must be only used with DefaultMessage type",
+      );
+    }
+
+    const solanaSignature = await signerContext(deviceId, signer => {
+      return signer.signMessage(
+        account.freshAddressPath,
+        (messageOptions as DefaultMessage).message,
+      );
+    });
```

Comment:
You might be able to get typescript to work here with this changes
```suggestion
    const message = messageOptions.message;
    if (!message || typeof message !== "string") {
      throw new Error(
        "Sign off-chain message on Solana must be only used with DefaultMessage type",
      );
    }

    const solanaSignature = await signerContext(deviceId, signer => {
      return signer.signMessage(
        account.freshAddressPath,
        message,
      );
    });
```

Reply:
done